sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -0,0 +1,326 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { AtomXmlSerializer, HttpOperationResponse } from "@azure/core-http";
+import * as Constants from "../util/constants";
+import { serializeToAtomXmlRequest, deserializeAtomXmlResponse } from "../util/atomXmlHelper";
+
+const requestProperties = ["Name", "Filter", "Action"];
+
+export function buildRuleOptions(name: string, ruleOptions: RuleOptions = {}): InternalRuleOptions {
+  const internalRuleOptions: InternalRuleOptions = Object.assign(ruleOptions, { name: name });
+  return internalRuleOptions;
+}
+
+export function buildRule(rawRule: any): Rule | {} {
+  if (rawRule == undefined || rawRule == {}) {
+    return {};
+  } else {
+    const result: Rule = {
+      ruleName: rawRule["RuleName"],
+      topicName: rawRule["TopicName"],
+      subscriptionName: rawRule["SubscriptionName"],
+      filter: getTopicFilterOrUndefined(rawRule["Filter"]),
+      action: rawRule["Action"],
+      createdAt: rawRule["CreatedAt"]
+    };
+    return result;
+  }
+}
+
+/**
+ *  @ignore
+ * Helper utility to retrieve `filter` value from given input or undefined,
+ * or undefined if not passed in.
+ * @param value
+ */
+function getTopicFilterOrUndefined(value: any): Filter | undefined {
+  if (value == undefined || value == {}) {
```

Comment:
I'm not sure `value == {}` is doing what you intend. For example, even `{} == {}` will return false, so this should never return true as far as I'm aware.

You could do something like `!Object.keys({}).length` if you just want to see if the object lacks any fields...do you have cases where an empty object gets passed to this function?

Reply:
I see, yes this seems unnecessary - updated the checks.
Also updated the Filter response structure to reflect one of sql Vs correlation filter types